프로젝트 API 구현

shkilo opened the pull request:
### 작업내용
1. 테스트코드 작성
2. 프로젝트 관련 모든 API 구현

### 코멘트
* task 생성할 때 dueDate를 어떻게 넣느냐에 따라서 쿼리부분이 달라져야 합니다.
* user 생성시 title이 관리함인 프로젝트를 생성해야합니다.
* 오늘에 해당하는 작업정보를 받아오는 부분은 task api 쪽으로 분리해도 될 것 같습니다.
* 프로젝트 개별 조회시 하위 task include
* 프로젝트 생성시 하위 섹션 하나 생성하는 부분 transaction 이용
* section 부분은 다른 파일로 분리하는것도 좋을것 같습니다

### 이슈번호
#21 #22 #23 #24 #25 #26 #27 #28 #29

shkilo committed:
[BE] FIX: jest module alias 문제 해결

shkilo committed:
[BE] TEST: project get api 테스트코드 작성

shkilo committed:
[BE] FEAT: 오늘 날짜의 시작과 끝을 반환하는 유틸 함수 작성

shkilo committed:
[BE] FEAT: project get all api 구현

+ sequelize.fn, group 이용 작업 개수 쿼리 구현
+ '오늘' 정보는 기한이 오늘까지인 작업의 개수를 쿼리하여 만듦

ed-jinyoung-park commented on server/src/utils/today-start-end.js:
```diff
@@ -0,0 +1,11 @@
+  return { todayStart, todayEnd };
+};
+
+module.exports = getTodayStartEnd;
```
제가 utils/date 파일을 만들어서 dueDate 검증하는 함수 넣어놨는데 date 관련 함수들을 utils/date로 모으는게 어떨까요?

shkilo replied:
좋을것같습니다!

ed-jinyoung-park commented on server/src/controllers/project.js:
```diff
@@ -0,0 +1,40 @@
+    errorHandler(res, 400, { message: err.message });
+  }
+};
+
```
말씀하신대로 ``/api/task/today``나 ``/api/project/today``를 엔드포인트로 오늘 task를 가져오는 API를 따로 분리해도 좋겠네요

ed-jinyoung-park commented:
title이 관리함인건 task들 중에 ``projectId``가 ``null``인것들을 담으면 되지 않을까요?

pjy0416 commented on server/src/controllers/project.js:
```diff
@@ -0,0 +1,40 @@
+
+    responseHandler(res, 201, projects);
+  } catch (err) {
+    errorHandler(res, 400, { message: err.message });
```
end point에 error handler가 작동하기 때문에, errorhandler를 직접 선언하기보다는 next(err)로 넘기는게 더 좋을 것 같습니다!!

pjy0416 commented on server/test/project.api.test.js:
```diff
@@ -0,0 +1,40 @@
+            throw err;
+          }
+
+          expect(res.body).toEqual(expectedProjects);
```
toStrictEqual를 사용해서 테스트하는게 더 적합하다고 생각합니다.
https://jestjs.io/docs/en/expect#tostrictequalvalue 를 참고해보면, 
toEqual에서는 {a: undefined, b: 2} 와 {b:2} 가 match된다고 판별될 것 같습니다!

shkilo replied:
문서 읽어보니 toStrictEqual 이 더 적합해 보입니다.

pjy0416 reviewed (approved):
고생하셨습니다!!

shkilo committed:
[BE] TEST: project 관련 API 테스트코드 작성

shkilo committed:
[BE] FEAT: project 관련 API 라우팅

shkilo committed:
[BE] FEAT: project 모델 변경-isFavorite 기본값 설정

shkilo committed:
[BE] FEAT: section 모델 변경-title nullable

shkilo committed:
[BE] FEAT: project API 컨트롤러 구현

+ 프로젝트 개별 조회시 하위 task include
+ 프로젝트 생성시 하위 섹션 하나 생성하는 부분 transaction 이용
+ section 부분은 다른 파일로 분리하는것도 좋을것 같습니다

shkilo committed:
[BE] REFACTOR: project controller 에러 핸들러 변경

ed-jinyoung-park committed:
Merge branch 'dev_web' into dev_web
